distributed/worker.py

```diff
@@ -1970,9 +1952,11 @@ async def gather_dep(self, worker, dep, deps, total_nbytes, cause=None):
 
                 # dep states may have changed before gather_dep runs
                 # if a dep is no longer in-flight then don't fetch it
-                deps = tuple(dep for dep in deps if self.dep_state.get(dep) == "flight")
+                deps_ts = [self.tasks[key] for key in deps]
+                deps_ts = tuple(ts for ts in deps_ts if ts.state == "flight")
+                deps = [d.key for d in deps_ts]
```

Comment:
This really makes you want variable scope in list comprehensions :)

Reply:
YES